outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -247,6 +255,73 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
         );
     }
 
+    private getNoCompanyInsightsSection = () => {
+        let NoInsightsSection = null;
+        const company = this.state.company as Company;
+        if (company.isEmpty()) {
+            if (this.props.partner.isAddedToDatabase() && !this.enriched) {
+                NoInsightsSection = (
+                    <div>v
+                        <div>
+                            {_t("No company attached to this contact")}
+                        </div>
+                        <div className="odoo-secondary-button" style={{margin: "8px auto auto auto"}}
+                             onClick={this.enrichCompanyForPartnerRequest}>
+                            {_t("Create a Company")}
+                        </div>
+                    </div>
+                );
+            } else {
+                if (!this.context.isConnected()) {
+                    NoInsightsSection = (
+                        <div>
+                            {_t("No company linked to this contact could be enriched")}
+                        </div>
+                    );
+                } else {
+                    NoInsightsSection = (
+                        <div>
+                            {_t("No company linked to this contact could be enriched or found in Odoo")}
+                        </div>
+                    );
+                }
+            }
+        } else {
+            if (this.enriched) {
+                NoInsightsSection = (<div>
+                    {_t("No insights where found for this company")}
+                </div>);
+            } else {
+                NoInsightsSection = (
+                    <div>
+                        {_t("No insights for this company")}
+                        <div className='odoo-secondary-button' style={{margin: "8px auto auto auto"}}
+                             onClick={this.updateCompanyRequest}>
```

Comment:
"updateCompanyRequest" could probably be named better? "enrichCompany" ?

Reply:
Yes indeed, better rename it to enrichAndUpdate, because we also have an enrichAndCreate method?